### ISSUE TITLE
Tests?

### ISSUE THREAD

**Issue by chlohal** [twibiral/obsidian-execute-code#121]
We're seeing a lot of bugs (#119, #109, #108, #95) that would be relatively easy to automatically test. If we could set up a test suite & use GitHub Actions's free tier to run it on every PR, I think that would be helpful to catch regressions

**Comment by twibiral** [comment]
You are completely right and tests are great! But testing the user interface will be a challenge to implement.

**Comment by chlohal** [comment]
True, it'll be a challenge! I read some articles on testing Electron apps, but most of them assume that you're the owner of the app, which doesn't apply for us.

**Comment by milan338** [comment]
This might be a plausible approach: https://forum.obsidian.md/t/standard-approach-for-writing-automated-end-to-end-tests-for-plugins/31535/2
